### PR TITLE
[WIP] THREESCALE-10233: Support OpenAPI v3.1

### PR DESCRIPTION
**Pull request by mayorova**
**What this PR does / why we need it**:

TBD

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-10233

**Verification steps** 

TBD

**Special notes for your reviewer**:

Looks like there are some incompatibilities with our current stack:
seems that `swagger-ui` v5 requires `babel-loader` v9 ([link](https://github.com/swagger-api/swagger-ui/blob/6940b97904c72d563ec410253d6326798d10d742/package.json#L126) which in its turn depends on `webpack` >=5, which requires `webpacker` v6.

So, I guess we need to park it until we upgrade webpack(er).

### CODE REVIEW

File: app/lib/three_scale/swagger/specification.rb [hunk 1]
```diff
@@ -67,6 +69,14 @@ def as_json
         end
       end
 
+      class V31 < V3x
+        JSON_SCHEMA = {'$ref' => 'https://spec.openapis.org/oas/3.1/schema/2022-10-07'}.freeze
+      end
+
+      class V30 < V3x
+        JSON_SCHEMA = {'$ref' => 'https://spec.openapis.org/oas/3.0/schema/2021-09-28'}.freeze
```
Comment (mayorova) [inline app/lib/three_scale/swagger/specification.rb:77]: Updated the previous link because it was broken

**Comment by mayorova** [comment]
Current blocker:


```
Error while parsing /home/dmayorov/Projects/3scale/porta/node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js
Line 2, column 439158: Hexadecimal digit expected.
`parseForESLint` from parser `/home/dmayorov/Projects/3scale/porta/node_modules/@typescript-eslint/parser/dist/index.js` is invalid and will just be ignored


ERROR in ./app/javascript/packs/active_docs.ts
Module Error (from ./node_modules/eslint-loader/dist/cjs.js):

/home/dmayorov/Projects/3scale/porta/app/javascript/packs/active_docs.ts
   2:23  error    Parse errors in imported module 'swagger-ui': Hexadecimal digit expected. (2:439158)   import/namespace
   2:23  error    Parse errors in imported module 'swagger-ui': Hexadecimal digit expected. (2:439158)   import/default
   2:23  warning  Parse errors in imported module 'swagger-ui': Hexadecimal digit expected. (2:439158)   import/no-named-as-default-member
  17:97  error    Unsafe argument of type `SwaggerUI.Request` assigned to a parameter of type `Request`  @typescript-eslint/no-unsafe-argument

✖ 4 problems (3 errors, 1 warning)


ERROR in ./app/javascript/packs/service_active_docs.ts
Module Error (from ./node_modules/eslint-loader/dist/cjs.js):

/home/dmayorov/Projects/3scale/porta/app/javascript/packs/service_active_docs.ts
  1:23  error    Parse errors in imported module 'swagger-ui': Hexadecimal digit expected. (2:439158)  import/namespace
  1:23  error    Parse errors in imported module 'swagger-ui': Hexadecimal digit expected. (2:439158)  import/default
  1:23  warning  Parse errors in imported module 'swagger-ui': Hexadecimal digit expected. (2:439158)  import/no-named-as-default-member

✖ 3 problems (2 errors, 1 warning)


ERROR in ./app/javascript/src/ActiveDocs/OAS3Autocomplete.ts
Module Error (from ./node_modules/eslint-loader/dist/cjs.js):

/home/dmayorov/Projects/3scale/porta/app/javascript/src/ActiveDocs/OAS3Autocomplete.ts
  14:81  error  Parse errors in imported module 'swagger-ui': Hexadecimal digit expected. (2:439158)  import/namespace

✖ 1 problem (1 error, 0 warnings)


ERROR in ./app/javascript/src/ActiveDocs/ThreeScaleApiDocs.ts
Module Error (from ./node_modules/eslint-loader/dist/cjs.js):

/home/dmayorov/Projects/3scale/porta/app/javascript/src/ActiveDocs/ThreeScaleApiDocs.ts
    1:23  error    Parse errors in imported module 'swagger-ui': Hexadecimal digit expected. (2:439158)                         import/namespace
    1:23  error    Parse errors in imported module 'swagger-ui': Hexadecimal digit expected. (2:439158)                         import/default
    1:23  warning  Parse errors in imported module 'swagger-ui': Hexadecimal digit expected. (2:439158)                         import/no-named-as-default-member
   11:38  error    Parse errors in imported module 'swagger-ui': Hexadecimal digit expected. (2:439158)                         import/namespace
  105:7   error    Variable name `RequestBodyTransformerPlugin` must match one of the following formats: camelCase, UPPER_CASE  @typescript-eslint/naming-convention

✖ 5 problems (4 errors, 1 warning)


ERROR in ./node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js 5507:11
Module parse failed: Unexpected token (5507:11)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
You may need an additional loader to handle the result of these loaders.
|     });
|   var operation_wrapper = (e, t) => class OperationWrapper extends Ke.default.Component {
>     onLoad = e => {
|       const {
|           operation: r
ℹ ｢wdm｣: Failed to compile.
```

File: app/lib/three_scale/swagger/specification.rb (hunk 1 above)
Comment (jlledom) [reply app/lib/three_scale/swagger/specification.rb:77]: Was there any consequences of this broken link?

**Comment by mayorova** [comment]
Closing until we upgrade to Webpacker (Shakapacker) 6 / Webpack 5

**Pull request closed without merging**